core/aws_ddk_core/stages/athena_sql.py

```diff
@@ -0,0 +1,156 @@
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        environment_id: str,
+        query_string: str,
+        workgroup: Optional[str] = None,
+        catalog_name: Optional[str] = None,
+        database_name: Optional[str] = None,
+        output_bucket_name: Optional[str] = None,
+        ouput_object_key: Optional[str] = None,
+        encryption_option: Optional[EncryptionOption] = None,
+        encryption_key: Optional[Key] = None,
+        state_machine_input: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        DDK Athena SQL stage.
+
+        Stage that contains a step function that executre Athena SQL query.
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the stage
+        environment_id : str
+            Identifier of the environment
+        query_string : str
+            SQL query that will be started
+        workgroup : Optional[str]
+            Workgroup name
+        catalog_name : Optional[str]
+            Catalog name
+        database_name : Optional[str]
+            Database name
+        output_bucket_name : Optional[str]
+            Output S3 bucket name
+        ouput_object_key : Optional[str]
+            Output S3 key
+        encryption_option : Optional[EncryptionOption]
+            Encryption configuration
+        encryption_key : Optional[Key]
+            Encryption KMS key
+        state_machine_input : Optional[Dict[str, Any]]
+            Input of the state machine
+        """
+        super().__init__(scope, id)
+
+        self._state_machine_input: Optional[Dict[str, Any]] = state_machine_input
+        self._event_detail_type: str = f"{id}-event-type"
+
+        # Create AthenaStartQueryExecution step function task
+        start_query_exec: AthenaStartQueryExecution = AthenaStartQueryExecution(
+            self,
+            "start-query-exec",
+            query_string=query_string,
+            integration_pattern=IntegrationPattern.RUN_JOB,
+            query_execution_context=QueryExecutionContext(
+                catalog_name=catalog_name,
+                database_name=database_name,
+            )
+            if catalog_name and database_name
+            else None,
+            result_configuration=ResultConfiguration(
+                encryption_configuration=EncryptionConfiguration(
+                    encryption_option=encryption_option,
+                    encryption_key=encryption_key,
+                )
+                if encryption_option
+                else None,
+                output_location=Location(
+                    bucket_name=output_bucket_name,
+                    object_key=ouput_object_key,
+                )
+                if output_bucket_name and ouput_object_key
+                else None,
+            )
+            if not workgroup
```

Comment:
Is the idea here that we should always use the workgroup if supplied instead of using the supplied result configuration? Would it work for a situation like UNLOAD like we discussed in the Wrangler issue (i.e. that you can bypass the Workgroup config if you want to)? I would prefer to delegate this to Athena instead perhaps

Reply:
You must either pass result config or workgroup